src/sqlfluff/core/rules/analysis/select_crawler.py

```diff
@@ -0,0 +1,141 @@
+"""Tools for more complex analysis of SELECT statements."""
+from collections import defaultdict
+from typing import Dict, List, NamedTuple, Optional, Union
+
+from cached_property import cached_property
+
+from sqlfluff.core.dialects.dialect_ansi import AliasInfo
+from sqlfluff.core.dialects.base import Dialect
+from sqlfluff.core.parser.segments.base import BaseSegment
+from sqlfluff.core.rules.analysis.select import get_select_statement_info
+
+
+class WildcardInfo(NamedTuple):
+    """Structure returned by SelectCrawler.get_wildcard_info()."""
+
+    segment: BaseSegment
+    tables: List[str]
+
+
+class SelectCrawler:
```

Comment:
The new structure is great, I think it makes it a lot clearer and it sets an example for future "statement specific crawling".
However there's a "BaseCrawler" in the project so as a contributor I'd be a little confused that other crawlers do not inherit from the "Base".

I'm not sure what the right answer is however, do you think in a follow up PR it would make sense to rename BaseCrawler to BaseRule? And keep the "crawler" concept to functions that browse the parsed SQL?

Reply:
I like that idea. I think calling them "rules" aligns better with the documentation.

I'll create a ticket for doing that.

Reply:
did you create a ticket? just making sure it's not lost

Reply:
I did. It's #840.